[Feature] 앱 시작 화면인 로그인 화면을 구현했습니다.

letusHyun opened the pull request:
🚨 **Your checklist for this pull request** 

- [x]  Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!

- [x]  Check the commit's or even all commits' message styles matches our requested structure.

## 📌 [요약]
- #42 
- 로그인 화면 UI 구현
- 앱 아이콘 추가
## ✨ [작업 내용]
- BaseLoginButton을 부모 클래스로 두고, 각각의 버튼은 자식클래스로 구현했습니다.
- panGesture를 사용해서 CircleView가 top 경계를 넘으면 maxTop경계로 위치시키고, bottom 경계를 넘으면 최초의 origin으로 위치시킵니다. 
- velocity사용해서 조금만 위로 드래그 하거나 아래로 드래그해도 circleView의 위치를 UX에 맞게 변화시켰습니다.
<img width="600" alt="image" src="https://github.com/IF-TG/iOS/assets/108918481/6b68e7c0-06fb-41ee-94c9-994143d51f17">


## 📸  [스크린샷(선택)]
<img width="250" alt="image" src="https://github.com/IF-TG/iOS/assets/108918481/34505156-57b7-4493-b96f-866b30dc16ca">

## 📚 TODO
- 각 플랫폼별 로그인 기능을 추가해야합니다.

letusHyun committed:
[Feat] 온보딩 이미지 및 동영상 추가

letusHyun committed:
[Feat] LoginStartView 추가

letusHyun committed:
[Feat] 온보딩 동영상 및 StartView 추가

letusHyun committed:
[Feat] arrow 아이콘 추가

letusHyun committed:
[Feat] 로그인 화면 구현 위해 임시 설정

letusHyun committed:
[Feat] 여가 아이콘 추가

letusHyun committed:
[Feat] 앱 아이콘 추가

letusHyun committed:
[Feat] 앱 아이콘 변경

letusHyun committed:
[Feat] 여가 로고 아이콘 변경

letusHyun committed:
[Style] 컨벤션에 맞게 manager에서 adapter로 네이밍 변경

letusHyun committed:
[Style] 네이밍변경에 따른 코드 수정

letusHyun committed:
[System] pbxproj

letusHyun committed:
[Feat] loginStartView 추가

letusHyun committed:
[Feat] top에 존재하는 여가 컴포넌트 추가

letusHyun committed:
Merge branch 'develop' of https://github.com/IF-TG/iOS into feature/onboarding

letusHyun committed:
[Chore] pod update

letusHyun committed:
[Feat] 로그인 버튼 활성화 시, startView 제거

letusHyun committed:
[Feat] NSRange에 특정 문자열이 포함되는지 판별하는 변수 추가

포함된다면 self를, 포함되지 않는다면 nil을 반환

letusHyun committed:
[System] pbxproj

letusHyun committed:
[Feat] 여가 description label 추가

letusHyun committed:
[Feat] 로그인 화면 애니메이션 적용

letusHyun committed:
[Feat] 복잡한 코드 함수화

letusHyun committed:
[Feat] LoginView 추가

letusHyun committed:
[Rebuild] LoginView 클래스를 정의해서 VC로부터 분리

letusHyun committed:
[System] pbxproj

letusHyun committed:
[Feat] BaseLoginButton을 공통뷰로 이동

letusHyun committed:
[Feat] 로그인 로고 추가

letusHyun committed:
[Rebuild] 불필요한 convenience init 제거

letusHyun committed:
[System] pbxproj

letusHyun committed:
[Feat] 각 기업별 로그인 버튼 추가

letusHyun committed:
[Feat] UIColor에 공통프로퍼티로 카카오 색상 추가

letusHyun committed:
[Feat] 각 버튼에 부가적인 UI 추가

letusHyun committed:
[Feat] BaseButton에 부가적인 UI 추가

letusHyun committed:
[System] pbxproj

letusHyun committed:
[Feat] 정책관련 label 추가 및 애니메이션 구현

letusHyun committed:
[Feat] 로그인 버튼 delegate 추가

letusHyun committed:
[Feat] 매직넘버 제거

letusHyun committed:
[Feat] 델리게이트 패턴으로 버튼의 input 연결

letusHyun committed:
[System] pbxproj

letusHyun committed:
[Remove] 불필요한 파일 제거

letusHyun committed:
[System] pbxproj

letusHyun committed:
[Feat] 클래스에 final 키워드 추가

letusHyun committed:
[Style] 여가 앱 아이콘 변경

letusHyun committed:
[Feat] 버튼 input ouput 처리

letusHyun committed:
[Style] 매직넘버를 Constant로 변경

letusHyun committed:
Merge branch 'develop' of https://github.com/IF-TG/iOS into feature/onboarding

# Conflicts:
#	travelPlan/travelPlan.xcodeproj/project.pbxproj

letusHyun committed:
[Style] 로그인 여부 변경

letusHyun committed:
[Rebuild] 의미 명확하게 전달할 수 있도록 변수 추가

SHcommit commented on travelPlan/Source/Feature/Authentication/Login/View/LoginView.swift:
```diff
@@ -0,0 +1,278 @@
+  }
+  
+  private func disappearFirstDescriptionLabel() {
+    UIView.animate(withDuration: 0.5,
+                   animations: {
+      self.firstDescriptionLabel.alpha = 0
+    }) { _ in
+      self.firstDescriptionLabel.snp.removeConstraints()
+      self.firstDescriptionLabel.removeFromSuperview()
+    }
+  }
```
오... : )

SHcommit commented on travelPlan/Source/Feature/Authentication/Login/View/LoginView.swift:
```diff
@@ -0,0 +1,278 @@
+        self?.loginStartView.snp.removeConstraints()
+        self?.loginStartView.removeFromSuperview()
+        
+        guard let self else { return }
```
여기서 왜 strong self를 선언했나요?????

SHcommit commented on travelPlan/Source/Feature/Authentication/Login/ViewController/LoginViewController.swift:
```diff
@@ -91,3 +112,64 @@ extension LoginViewController: ViewBindCase {
+}
+
+// MARK: - LayoutSupport
+extension LoginViewController: LayoutSupport {
+  func addSubviews() {
+    view.addSubview(loginView)
+  }
+  
+  func setConstraints() {
+    loginView.snp.makeConstraints {
+      $0.edges.equalToSuperview()
+    }
+  }
+}
```
```
override func loadView() {
  view = loginView
}
```

이렇게 하는건 어떨가요?!?!??

SHcommit commented on travelPlan/Source/Feature/Authentication/Login/View/LoginStartView.swift:
```diff
@@ -0,0 +1,255 @@
+  
+  private func animateCircleView(velocity: CGPoint) {
+    let currentCircleViewX = circleView.frame.origin.x
+    var isDraggingDown = velocity.y > .zero
```
제스처 터치가 끝났고, 감속이 살짝만 있어도 circleView가 위로 가도록 설정하신 것 맞나요?..?

SHcommit commented on travelPlan/Source/Feature/Authentication/Login/View/LoginView.swift:
```diff
@@ -0,0 +1,278 @@
+        self?.backgroundColor = .black.withAlphaComponent(0.24)
+        self?.loginStartView.alpha = 0
+      }) { _ in
+        self?.loginStartView.snp.removeConstraints()
+        self?.loginStartView.removeFromSuperview()
```
스크롤 후 사용하지 않는 뷰 제거하는게 좋네요 +_+

letusHyun replied:
감사합니다!

SHcommit reviewed (approved):
자연스럽고 커스텀 화면 전환이 정말 좋네요. 마음에드는 애니메이션입니다 +_+ 
수고하셧습니다.

letusHyun committed:
[Rebuild] 불필요한 코드 변경

letusHyun committed:
[Style] 요구사항에 맞게 text 변경

letusHyun committed:
[Rebuild] 불필요한 guard let 구문 제거

letusHyun committed:
Merge branch 'develop' of https://github.com/IF-TG/iOS into feature/onboarding

# Conflicts:
#	travelPlan/Podfile.lock
#	travelPlan/Pods/Alamofire/README.md
#	travelPlan/Pods/Alamofire/Source/AuthenticationInterceptor.swift
#	travelPlan/Pods/Alamofire/Source/Combine.swift
#	travelPlan/Pods/Alamofire/Source/MultipartUpload.swift
#	travelPlan/Pods/Alamofire/Source/NetworkReachabilityManager.swift
#	travelPlan/Pods/Alamofire/Source/Protected.swift
#	travelPlan/Pods/Alamofire/Source/Request.swift
#	travelPlan/Pods/Alamofire/Source/ResponseSerialization.swift
#	travelPlan/Pods/Alamofire/Source/URLEncodedFormEncoder.swift
#	travelPlan/Pods/Manifest.lock
#	travelPlan/Pods/Pods.xcodeproj/project.pbxproj
#	travelPlan/Pods/Target Support Files/Alamofire/Alamofire-Info.plist
#	travelPlan/travelPlan.xcodeproj/project.pbxproj

letusHyun committed:
Merge branch 'develop' of https://github.com/IF-TG/iOS into feature/onboarding

# Conflicts:
#	travelPlan/travelPlan.xcodeproj/project.pbxproj

letusHyun committed:
[System] pbxproj

letusHyun committed:
[Comment] 주석 변경

letusHyun committed:
[Feat] 로그인 화면 동영상을 관리하는 Manager객체 추가

LoginPlayer를 관리해주는 Manager 싱글톤 객체를 구현했습니다.
setupPlayer(in:)메소드를 사용하면 동영상이 자동으로 실행됩니다.
동영상을 더 이상 실행하지 않는 경우는 명시적으로 cleadup()메소드를 사용해서 불필요한 리소스를 제거해야 합니다.

letusHyun committed:
[Rebuild] Manager객체 생성으로 인해 불필요한 코드 제거

letusHyun committed:
[System] pbxproj

letusHyun committed:
[Fix] 앱을 나왔다가 다시 들어갔을 때 동영상이 재생되지 않는 버그 수정

letusHyun committed:
[Feat] LoginViewController인 경우에만 동영상이 재생되도록 추가

SHcommit reviewed (approved):
확인했습니다: )

SHcommit commented on travelPlan/Source/Common/Manager/LoginPlayerManager.swift:
```diff
@@ -0,0 +1,84 @@
+import UIKit
+import Combine
+
+final class LoginPlayerManager {
```
LoginViewController에서만 사용되는데 싱글톤으로 관리하는 이유가 궁금합니다!!!!
